ang/civicase/contact/directives/contact-card.directive.js

```diff
@@ -74,11 +74,17 @@
         if (typeof $scope.data === 'string') {
           return ContactsCache.add([$scope.data]);
         } else {
-          var contactIDs = $scope.data.map(function (contact) {
-            if (contact) {
-              return contact.contact_id;
-            }
-          });
+          var contactIDs;
+
+          if (_.isPlainObject($scope.data)) {
+            contactIDs = _.keys($scope.data);
+          } else {
+            contactIDs = $scope.data.map(function (contact) {
+              if (contact) {
+                return contact.contact_id;
+              }
+            });
+          }
 
           return ContactsCache.add(contactIDs);
```

Comment:
```js
function fetchContactsInfo () {
  var contactIds;

  if (typeof $scope.data === 'string') {
    contactIds = [ $scope.data ];
  } else if (_.isPlainObject($scope.data)) {
    contactIds = _.keys($scope.data);
  } else {
    contactIds = _.chain($scope.data)
      .compact() // <-------- Note: this removes falsy values. But why are there falsy values anyways? 🤦 
      .map('contact_id')
      .value();
   }

  return ContactsCache.add(contactIds);
}
```

Reply:
> Note: this removes falsy values. But why are there falsy values anyways? 🤦

I have not fixed this yet, because we need refactoring of the whole contact card, so when we do it, this will be taken care of.

Reply:
Yes, more weird things from the parameters problem.